index/scorch/snapshot_index.go

```diff
@@ -464,9 +469,9 @@ func (i *IndexSnapshot) recycleTermFieldReader(tfr *IndexSnapshotTermFieldReader
 	}
 	i.fieldTFRs[tfr.field] = append(i.fieldTFRs[tfr.field], tfr)
 	if i.fieldDicts == nil {
-		i.fieldDicts = map[string][]segment.TermDictionary{}
+		i.fieldDicts = map[string][][]segment.TermDictionary{}
 	}
-	i.fieldDicts[tfr.field] = tfr.dicts
+	i.fieldDicts[tfr.field] = append(i.fieldDicts[tfr.field], tfr.dicts)
```

Comment:
Looks like the tfr has a dicts field, so one "localized, peephole" thought is I'm wondering if tfr.dicts needs to be cleared?  That way, if a future user picks up the tfr from the i.fieldTFRs, the dicts ought to be empty?

A second, larger thought, is perhaps we only need reuse of i.fieldTFRs and don't need the extra fieldDicts?  That is, if a tfr has a dicts field, the perhaps reuse of tfr's is enough?  Not sure if that's right, but perhaps, @abhinavdangeti, can you look into this and see if that simplification might work?

Reply:
Hmm, it seems like that simplification might just work, let me look into the code a bit.